rules/stamp.bzl

```diff
@@ -0,0 +1,82 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Utilities to detect whether --stamp was specified on the command-line.
+# This relies on the workaround explained in
+# https://github.com/bazelbuild/bazel/issues/11164#issuecomment-617336309
+
+# Return a rule attribute to add a stamp parameters. This stamping parameter follows
+# the usual bazel convention:
+# - `stamp = 1` means that stamping information is available, even if stamping is
```

Comment:
Can we use string names to avoid magic numbers?

Reply:
Do you mean avoid the values `0`, `1` and `-1`? If so, that would break an established bazel convention on the value of the `stamp` parameter which I am not sure is really helpful (although I agree that this convention is not optimal).

Reply:
Ah, makes sense.